src/AudioTrackManager.ts

```diff
@@ -1,85 +1,156 @@
 import { AudioTrack } from "./types";
-import MediaAudio from "./MediaAudio";
+
+interface Constructable<T> {
+  new (): T;
+}
+
+interface WindowWithAudioContext extends Window {
+  AudioContext?: Constructable<AudioContext>;
+  webkitAudioContext?: Constructable<AudioContext>;
+}
+
+declare const window: WindowWithAudioContext;
 
 class AudioTrackManager {
+  private audioContext: AudioContext | undefined;
+
   public isPlaying: boolean;
 
-  private currentAudio: MediaAudio[];
+  private currentAudio: {
+    source: AudioBufferSourceNode;
+    gainNode: GainNode;
+    url: string;
+    originalVolume: number;
+  }[];
+
+  private muted = false;
 
   constructor() {
     this.isPlaying = false;
     this.currentAudio = [];
   }
 
-  public play(audioTracks: AudioTrack[]): void {
-    if (audioTracks.length === 0) return;
+  private async loadAudioBuffer(url: string): Promise<AudioBuffer | undefined> {
+    if (this.audioContext === undefined) return undefined;
 
-    this.isPlaying = true;
+    const response = fetch(url);
```

Comment:
I think usually response = await fetch
Saves the funky await (await on the following line